2025-03-10 01:02:52,419 INFO    Thread-7 (_run_job):33948 [wandb_setup.py:_flush():67] Current SDK version is 0.19.8
2025-03-10 01:02:52,419 INFO    Thread-7 (_run_job):33948 [wandb_setup.py:_flush():67] Configure stats pid to 33948
2025-03-10 01:02:52,419 INFO    Thread-7 (_run_job):33948 [wandb_setup.py:_flush():67] Loading settings from /home/dextrolaev/.config/wandb/settings
2025-03-10 01:02:52,419 INFO    Thread-7 (_run_job):33948 [wandb_setup.py:_flush():67] Loading settings from /home/dextrolaev/Desktop/DL/da6401_assignment1/wandb/settings
2025-03-10 01:02:52,419 INFO    Thread-7 (_run_job):33948 [wandb_setup.py:_flush():67] Loading settings from environment variables
2025-03-10 01:02:52,419 INFO    Thread-7 (_run_job):33948 [wandb_init.py:setup_run_log_directory():647] Logging user logs to /home/dextrolaev/Desktop/DL/da6401_assignment1/wandb/run-20250310_010252-7rsio6eb/logs/debug.log
2025-03-10 01:02:52,419 INFO    Thread-7 (_run_job):33948 [wandb_init.py:setup_run_log_directory():648] Logging internal logs to /home/dextrolaev/Desktop/DL/da6401_assignment1/wandb/run-20250310_010252-7rsio6eb/logs/debug-internal.log
2025-03-10 01:02:52,419 INFO    Thread-7 (_run_job):33948 [wandb_init.py:init():761] calling init triggers
2025-03-10 01:02:52,419 INFO    Thread-7 (_run_job):33948 [wandb_init.py:init():766] wandb.init called with sweep_config: {'activation_function': 'sigmoid', 'batch_size': 32, 'epochs': 10, 'hidden_layer_size': 128, 'learning_rate': 0.0001, 'num_of_hidden_layers': 3, 'optimizer': 'momentum', 'weight_decay': 0, 'weight_initialization': 'xavier'}
config: {'_wandb': {}}
2025-03-10 01:02:52,419 INFO    Thread-7 (_run_job):33948 [wandb_init.py:init():784] starting backend
2025-03-10 01:02:52,419 INFO    Thread-7 (_run_job):33948 [wandb_init.py:init():788] sending inform_init request
2025-03-10 01:02:52,420 INFO    Thread-7 (_run_job):33948 [backend.py:_multiprocessing_setup():101] multiprocessing start_methods=fork,spawn,forkserver, using: spawn
2025-03-10 01:02:52,420 INFO    Thread-7 (_run_job):33948 [wandb_init.py:init():798] backend started and connected
2025-03-10 01:02:52,420 INFO    Thread-7 (_run_job):33948 [wandb_run.py:_config_callback():1261] config_cb None None {'activation_function': 'sigmoid', 'batch_size': 32, 'epochs': 10, 'hidden_layer_size': 128, 'learning_rate': 0.0001, 'num_of_hidden_layers': 3, 'optimizer': 'momentum', 'weight_decay': 0, 'weight_initialization': 'xavier'}
2025-03-10 01:02:52,421 INFO    Thread-7 (_run_job):33948 [wandb_init.py:init():891] updated telemetry
2025-03-10 01:02:52,425 INFO    Thread-7 (_run_job):33948 [wandb_init.py:init():915] communicating run to backend with 90.0 second timeout
2025-03-10 01:02:52,936 INFO    Thread-7 (_run_job):33948 [wandb_init.py:init():990] starting run threads in backend
2025-03-10 01:02:52,997 INFO    Thread-7 (_run_job):33948 [wandb_run.py:_console_start():2375] atexit reg
2025-03-10 01:02:52,997 INFO    Thread-7 (_run_job):33948 [wandb_run.py:_redirect():2227] redirect: wrap_raw
2025-03-10 01:02:52,998 INFO    Thread-7 (_run_job):33948 [wandb_run.py:_redirect():2292] Wrapping output streams.
2025-03-10 01:02:52,998 INFO    Thread-7 (_run_job):33948 [wandb_run.py:_redirect():2315] Redirects installed.
2025-03-10 01:02:52,998 INFO    Thread-7 (_run_job):33948 [wandb_init.py:init():1032] run started, returning control to user process
2025-03-10 01:02:53,001 INFO    Thread-7 (_run_job):33948 [wandb_run.py:_finish():2112] finishing run cs24s031-indian-institute-of-technology-madras/dl-assignment1/7rsio6eb
2025-03-10 01:02:53,001 INFO    Thread-7 (_run_job):33948 [wandb_run.py:_atexit_cleanup():2340] got exitcode: 1
2025-03-10 01:02:53,001 INFO    Thread-7 (_run_job):33948 [wandb_run.py:_restore():2322] restore
2025-03-10 01:02:53,001 INFO    Thread-7 (_run_job):33948 [wandb_run.py:_restore():2328] restore done
2025-03-10 01:02:55,953 INFO    MsgRouterThr:33948 [mailbox.py:close():129] Closing mailbox, abandoning 1 handles.
2025-03-10 01:02:55,954 INFO    Thread-7 (_run_job):33948 [wandb_run.py:_restore():2322] restore
2025-03-10 01:02:55,954 INFO    Thread-7 (_run_job):33948 [wandb_run.py:_restore():2328] restore done
2025-03-10 01:02:55,954 ERROR   Thread-7 (_run_job):33948 [wandb_run.py:_atexit_cleanup():2361] Problem finishing run
Traceback (most recent call last):
  File "/home/dextrolaev/myenv/lib/python3.12/site-packages/wandb/agents/pyagent.py", line 306, in _run_job
    self._function()
  File "/home/dextrolaev/Desktop/DL/da6401_assignment1/neural_net2.py", line 614, in train
    nn.train(
  File "/home/dextrolaev/Desktop/DL/da6401_assignment1/neural_net2.py", line 517, in train
    dw, db = self.backpropagation(a_li, h_li, y_hat_i, batch_label, batch_data)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/dextrolaev/Desktop/DL/da6401_assignment1/neural_net2.py", line 486, in backpropagation
    grad_L_hi = np.dot(dA[-1], self.W[i].T)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: shapes (32,10) and (128,10) not aligned: 10 (dim 1) != 128 (dim 0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/dextrolaev/myenv/lib/python3.12/site-packages/wandb/sdk/wandb_run.py", line 2352, in _atexit_cleanup
    self._on_finish()
  File "/home/dextrolaev/myenv/lib/python3.12/site-packages/wandb/sdk/wandb_run.py", line 2628, in _on_finish
    internal_messages_handle = self._backend.interface.deliver_internal_messages()
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/dextrolaev/myenv/lib/python3.12/site-packages/wandb/sdk/interface/interface.py", line 963, in deliver_internal_messages
    return self._deliver_internal_messages(internal_message)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/dextrolaev/myenv/lib/python3.12/site-packages/wandb/sdk/interface/interface_shared.py", line 479, in _deliver_internal_messages
    return self._deliver_record(record)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/dextrolaev/myenv/lib/python3.12/site-packages/wandb/sdk/interface/interface_shared.py", line 389, in _deliver_record
    handle = mailbox.require_response(record)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/dextrolaev/myenv/lib/python3.12/site-packages/wandb/sdk/mailbox/mailbox.py", line 68, in require_response
    raise MailboxClosedError()
wandb.sdk.mailbox.mailbox.MailboxClosedError
